### PR TITLE
change PreserveFullCustomRegistryPath in CRD yaml to be consistent wi…

### PR DESCRIPTION
**Pull request by haibinxie**
…th spec in code

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: deploy/crds/core_v1_storagecluster_crd.yaml [hunk 1]
```diff
@@ -96,7 +96,7 @@ spec:
                   Custom container image registry server that will be used instead of
                   index.docker.io to download Docker images. This may include the repository as well.
                   (Example: myregistry.net:5443 or myregistry.com/myrepository)
-              preserveFullCustomImageRegistry:
+              preserveFullCustomRegistryPath:
```
Comment (piyush-nimbalkar) [inline deploy/crds/core_v1_storagecluster_crd.yaml:99]: To be consistent with `customImageRegistry` can we call this `preserveFullCustomImageRegistry` or `preserveFullCustomImageRegistryPath`? Not sure if we will have a different registry than image registry in future, but it is better to be safe as this is a user facing field.
I know this involves more changes in the code, but better get this done once.

Comment (haibinxie) [reply deploy/crds/core_v1_storagecluster_crd.yaml]: Sounds good, changed.

File: pkg/apis/core/v1/storagecluster.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ type StorageClusterSpec struct {
 	// This is a hack to stop GetImageURN from swallowing part of the image tag.
 	// Without it having a customImageRegistry with a / in it means that an image
 	// `portworx/oci-monitor` becomes just `oci-monitor`
-	PreserveFullCustomRegistryPath bool `json:"preserveFullCustomRegistryPath"`
+	PreserveFullCustomImageRegistry bool `json:"preserveFullCustomImageRegistry"`
```
Comment (piyush-nimbalkar) [inline pkg/apis/core/v1/storagecluster.go:80]: Let's also add `omitempty` here.

**Review by piyush-nimbalkar — APPROVED** [review]

File: pkg/apis/core/v1/storagecluster.go (hunk 2 above)
Comment (haibinxie) [reply pkg/apis/core/v1/storagecluster.go]: great catch, updated.

**Pull request merged**